Market-Trading-System/src/test/java/UnitTests/UserFacadeUT.java

```diff
@@ -0,0 +1,136 @@
+package UnitTests;
+
+import DomainLayer.Market.IRepository;
+import DomainLayer.Market.User.IUserFacade;
+import DomainLayer.Market.User.ShoppingCart;
+import DomainLayer.Market.User.UserController;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+public class UserFacadeUT {
+
+    @Mock
+    private IRepository<Long, ShoppingCart> carts;
+
+    @Mock
+    private IRepository<Long, User> users;
+
+    @InjectMocks
+    private IUserFacade userFacade = new UserController();
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this); // Initializes annotated fields
+    }
+
+    @Test
+    void testRegister() {
+        // Arrange
+        String username = "newUser";
```

Comment:
put all the repetitive strings as final strings at the start of the class

Reply:
Resolved